### PR TITLE
Domains: Add notice when canceling or transferring a free Gravatar domain

### PR DESCRIPTION
**Pull request by leonardost**
## Proposed Changes

This PR adds notices and warnings explaining that users won't be able to get another free Gravatar domain in these situations:
- During checkout for a free Gravatar domain (via the `/start/domain-for-gravatar` flow)
- When transferring a free Gravatar domain away and
- When deleting a free Gravatar domain

### Screenshots

| Page | Before | After |
| --- | --- | --- |
| Checkout | Markup on 2024-06-14 at 20:35:50 | Markup on 2024-06-17 at 15:34:15 |
| Outbound transfer | Screenshot 2024-06-13 at 18 24 15 | Screenshot 2024-06-17 at 15 35 31 |
| Disable domain auto-renew | Screenshot 2024-06-13 at 16 05 19 | Markup on 2024-06-17 at 15:36:01 |
| Delete domain (after turning auto-renew off) | Screenshot 2024-06-13 at 16 12 29 | Markup on 2024-06-17 at 15:36:29 |

## Why are these changes being made?

- This is part of the Custom domains on Gravatar project (pcYYhz-24z-p2).

## Testing Instructions

- Open the live Calypso link or build this branch locally
- Checkout
  - Go to the domains for Gravatar flow (`/start/domain-for-gravatar`) and add a domain to the cart
  - Ensure the explanation about only one free Gravatar domain is shown in the checkout page
  - Visit the other flows (`/start`, adding domains to an existing site, etc) and ensure the free Gravatar domain copy isn't shown in the checkout page
- Outbound transfer
  - When you already have a free Gravatar domain, go to its domain transfer page and ensure a warning notice is shown
- Cancel domain
  - When you already have a free Gravatar domain, go to the domain cancelation page and ensure a warning text is shown
  - After you disable auto-renew for that domain, try to cancel it and ensure the deletion confirmation dialog also contains warning text explaining that you won't be able to get another free Gravatar domain

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ class RemoveDomainDialog extends Component {
 	};
 
 	renderDomainDeletionWarning( productName ) {
-		const { translate, slug, currentRoute } = this.props;
+		const { translate, slug, currentRoute, isGravatarDOmain } = this.props;
```
Comment (jjchrisdiehl) [inline client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: `isGravatarDOmain` typo?

File: client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx [hunk 2]
```diff
@@ -44,6 +44,13 @@ class RemoveDomainDialog extends Component {
 						}
 					) }
 				</p>
+				{ isGravatarDOmain && (
```
Comment (jjchrisdiehl) [inline client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: Typo here too

File: client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx [hunk 3]
```diff
@@ -267,6 +274,7 @@ export default connect( ( state, ownProps ) => {
 	const selectedDomainName = getName( ownProps.purchase );
 	const selectedDomain = getSelectedDomain( { domains, selectedDomainName } );
 	return {
+		isGravatarDOmain: selectedDomain.isGravatarDomain,
```
Comment (jjchrisdiehl) [inline client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: Here too

**Comment by jjchrisdiehl** [comment]
Overall looks good to me - only noted a typo that should be fixed 👍

**Review by hambai — COMMENTED** [review]
So besides the typo my concern here is the "free Gravatar domain" in the copy. Would it make sense to change that to "your complimentary Gravatar profile domain" or "free domain for your Gravatar profile"?

**So in the checkout:**
> This is a free Gravatar domain. You can only get one free domain to use with Gravatar

would become

> This is a complimentary Gravatar profile domain. You're eligible to only one free domain for your Gravatar profile.

**In outbound transfer:**

> This is a free Gravatar domain. If you transfer it to another registrar, you won't be able to get another free domain to use with Gravatar.

would become

> This is a complimentary Gravatar profile domain. You won't be able to get another free domain in case you transfer this one to another registrar.

**In Disable domain auto-renew:**

> This is a free Gravatar domain. If it expires, you won't be able to get another free domain to use with Gravatar.

would become

> This is a complimentary Gravatar profile domain. If it expires, you won't be able to get another free domain to use for your Gravatar profile

**And in Delete domain:**

> Also, this is a free Gravatar domain. If if is deleted, you won't be able to get another free domain to use with Gravatar.

would become

> Also, this is a complimentary Gravatar profile domain. If if is deleted, you won't be able to get another free domain to use with your Gravatar profile.

I'd prefer to get a second look by a native speaker though

File: client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx [hunk 4]
```diff
@@ -44,6 +44,13 @@ class RemoveDomainDialog extends Component {
 						}
 					) }
 				</p>
+				{ isGravatarDOmain && (
+					<p>
+						{ translate(
+							"Also, this is a free Gravatar domain. If if is deleted, you won't be able to get another free domain to use with Gravatar."
```
Comment (rafaelgallani) [inline client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: Small typo
```suggestion
							"Also, this is a free Gravatar domain. If it is deleted, you won't be able to get another free domain to use with Gravatar."
```

File: client/my-sites/checkout/src/components/domain-registration-free-gravatar-domain.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+	const domains = getDomainRegistrations( props.cart );
+
+	if ( ! domains.some( ( domain ) => domain.extra?.is_gravatar_domain ) ) {
+		return null;
+	}
+
+	return (
+		<CheckoutTermsItem>
+			{ translate(
+				'This is a free Gravatar domain. You can only get one free domain to use with Gravatar'
```
Comment (rafaelgallani) [inline client/my-sites/checkout/src/components/domain-registration-free-gravatar-domain.tsx]: Maybe add a period to match all other checkout messages/terms?
```suggestion
				'This is a free Gravatar domain. You can only get one free domain to use with Gravatar.'
```

**Review by rafaelgallani — COMMENTED** [review]
Just some small suggestions

**Comment by rafaelgallani** [comment]
> So besides the typo my concern here is the "free Gravatar domain" in the copy. Would it make sense to change that to "your complimentary Gravatar profile domain" or "free domain for your Gravatar profile"?

I'm not a native speaker, but I agree with Kamen's suggestion, although I prefer the latter — it feels more natural, and the translation seems better... And we still stand out the fact that the domain was given for free.

**Comment by leonardost** [comment]
Thanks for the review @jjchrisdiehl, @hambai and @rafaelgallani! I've fixed the typo and updated the copy with your suggestions. I'll ask @delputnam for a native speaker review 🙂

File: client/me/purchases/cancel-purchase/refund-information.jsx [hunk 6]
```diff
@@ -287,10 +290,20 @@ const CancelPurchaseRefundInformation = ( {
...
+			i18n.translate(
+				'When you cancel your domain, it will remain registered and active until the registration expires, ' +
+					'at which point it will be automatically removed from your site.'
+			),
+		];
+
+		if ( isGravatarDomain ) {
+			text.push(
+				i18n.translate(
+					"This is a free domain for your Gravatar profile. If it expires, you won't be able to get another free domain to use for your Gravatar profile."
```
Comment (delputnam) [inline client/me/purchases/cancel-purchase/refund-information.jsx]: This is a bit wordy, but let me know what you think:

> This domain, for use with your Gravatar profile, is provided at no cost for the first year. Please be aware that this offer is limited to one free domain per user. If you cancel this domain, you will have to pay the regular price to register another domain.

File: client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx [hunk 7]
```diff
@@ -44,6 +44,13 @@ class RemoveDomainDialog extends Component {
 						}
 					) }
 				</p>
+				{ isGravatarDomain && (
+					<p>
+						{ translate(
+							"Also, this is a free domain for your Gravatar profile. If it is deleted, you won't be able to get another free domain to use with your Gravatar profile."
```
Comment (delputnam) [inline client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx]: How about this?

This domain is provided at no cost for the first year for use with your Gravatar profile. This offer is limited to one per user. If you cancel this domain, you will have to pay the standard price to register another domain for your Gravatar profile.

File: client/my-sites/checkout/src/components/domain-registration-free-gravatar-domain.tsx [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+	const domains = getDomainRegistrations( props.cart );
+
+	if ( ! domains.some( ( domain ) => domain.extra?.is_gravatar_domain ) ) {
+		return null;
+	}
+
+	return (
+		<CheckoutTermsItem>
+			{ translate(
+				"This is a free domain for your Gravatar profile. You're eligible to only one free domain for your Gravatar profile."
```
Comment (delputnam) [inline client/my-sites/checkout/src/components/domain-registration-free-gravatar-domain.tsx]: > This domain is provided at no cost for the first year for use with your Gravatar profile. This offer is limited to one per user. If you cancel this domain, you will have to pay the standard price to register another domain for your Gravatar profile.

File: client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx [hunk 9]
```diff
@@ -372,6 +373,14 @@ const TransferPage = ( props: TransferPageProps ) => {
 						? __( 'Transfer to another registrar' )
 						: __( 'Advanced Options' ) }
 				</CardHeading>
+				{ domain?.isGravatarDomain && (
+					<InfoNotice
+						redesigned
+						text={ __(
+							"This is a free domain for your Gravatar profile. If you transfer it to another registrar, you won't be able to get another free domain to use with your Gravatar profile."
```
Comment (delputnam) [inline client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx]: > This domain is provided at no cost for the first year for use with your Gravatar profile. This offer is limited to one per user. If you transfer it to another registrar, you will have to pay the standard price to register another domain for your Gravatar profile.

**Review by delputnam — APPROVED** [review]
Let's try the language I've suggested and see how that looks.
